packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -284,6 +347,24 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
     }
     return new Stack(children: children);
   }
+
+  void _onProgressChanged() {
+    if (_previousAnimation.status != AnimationStatus.dismissed) {
+      _updateGeometryScale(_previousAnimation.value);
+      return;
+    }
+    if (_currentAnimation.status != AnimationStatus.dismissed) {
+      _updateGeometryScale(_currentAnimation.value);
+      return;
+    }
+  }
+
+  void _updateGeometryScale(double scale) {
+    widget.geometryNotifier.value = ScaffoldGeometry._update(
```

Comment:
this is going to cause the notifier to notify multiple times per layout, which seems suboptimal.

Reply:
we discussed this and couldn't find a workaround.